lib/ansible/modules/cloud/amazon/ecs_taskdefinition.py

```diff
@@ -174,12 +193,21 @@ def register_task(self, family, task_role_arn, network_mode, container_definitio
 
             validated_containers.append(container)
 
+        params = dict(
+            family=family,
+            taskRoleArn=task_role_arn,
+            networkMode=network_mode,
+            containerDefinitions=container_definitions,
+            volumes=volumes,
+            requiresCompatibilities=[launch_type]
```

Comment:
This doesn't seem to have any checks for whether boto3 supports `requiresCompatibilities`, which seems to only come with version 1.8.4 and beyond of botocore.

You'll need a check similar to https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/ecs_service.py#L443-L445

`requiresCompatibilities` should only be added to the parameter list if it's actually required.